### ISSUE TITLE
`@grad_from_chainrules` hygiene: cannot use custom types in method signature

### ISSUE THREAD

**Issue by manuelbb-upb** [JuliaDiff/ReverseDiff.jl#224]
As noticed somewhere in https://github.com/FluxML/NNlib.jl/issues/484, it appears that `@grad_from_chainrule` is not able to see custom types in method signatures. I looked through the issues here and hope I did not overlook it. 
My guess is that method argument expressions should be escaped, [here](https://github.com/JuliaDiff/ReverseDiff.jl/blob/f06b776333dfd3b5259a27bdfb789d8f017c1ee1/src/macros.jl#L320) or in the parsing later on.

MWE
```julia
using ReverseDiff
struct myNumber val end

# define chain rules here ...
ReverseDiff.@grad_from_chainrules sin(x::myNumber)
```

This gives `ERROR: UndefVarError: myNumber not defined`.
If we look into `@macroexpand ...` we see stuff like `sin(var"#54#x"::ReverseDiff.myNumber) `.

**Comment by avik-pal** [comment]
+1 for this. I had to solve this by manually defining the functions in https://github.com/LuxDL/LuxLib.jl/blob/65337c4b5cb8c49cca5472e9aad9e3d69d51e553/ext/LuxLibReverseDiffExt.jl#L50-L108

**Comment by mohamed82008** [comment]
fixed by #232
